app/controllers/checkouts_controller.rb

```diff
@@ -3,13 +3,26 @@ class CheckoutsController < ApplicationController
 
   def create
     @book = Book.find(params[:book_id])
-    checkout = Checkout.create(book: @book, user: current_user)
+    if CheckOutBook.process(book: @book, user: current_user)
+      flash[:success] = t('.successful_message')
```

Comment:
`redirect_to` accepts options for setting the flash. By default, it supports the `notice` and `alert` keys. We're using `success` and `error` here, so we can register those using `add_flash_types` in `ApplicationController`: http://api.rubyonrails.org/classes/ActionController/Flash/ClassMethods.html

Once that's done setting the flash and redirecting can be a one liner everywhere, which is nice.